TypeScript undefined-argument error in chrome-finder.ts

I'm getting the following errors when compiling my project:

```
59     priorities.push({regex: new RegExp(process.env.LIGHTHOUSE_CHROMIUM_PATH), weight: 150});
                                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/chrome-launcher/chrome-finder.ts(59,40): error TS2345: Argument of type 'string | undefined' is not assignable to parameter of type 'string'.
  Type 'undefined' is not assignable to type 'string'.


63     priorities.push({regex: new RegExp(process.env.CHROME_PATH), weight: 151});
                                          ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/chrome-launcher/chrome-finder.ts(63,40): error TS2345: Argument of type 'string | undefined' is not assignable to parameter of type 'string'.
  Type 'undefined' is not assignable to type 'string'.


71   if (canAccess(process.env.CHROME_PATH)) {
                   ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/chrome-launcher/chrome-finder.ts(71,17): error TS2345: Argument of type 'string | undefined' is not assignable to parameter of type 'string'.
  Type 'undefined' is not assignable to type 'string'.


75   if (canAccess(process.env.LIGHTHOUSE_CHROMIUM_PATH)) {
                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/chrome-launcher/chrome-finder.ts(75,17): error TS2345: Argument of type 'string | undefined' is not assignable to parameter of type 'string'.
  Type 'undefined' is not assignable to type 'string'.


139     priorities.push({regex: new RegExp(process.env.LIGHTHOUSE_CHROMIUM_PATH), weight: 100});
                                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/chrome-launcher/chrome-finder.ts(139,40): error TS2345: Argument of type 'string | undefined' is not assignable to parameter of type 'string'.
  Type 'undefined' is not assignable to type 'string'.


143     priorities.push({regex: new RegExp(process.env.CHROME_PATH), weight: 101});
                                           ~~~~~~~~~~~~~~~~~~~~~~~

node_modules/chrome-launcher/chrome-finder.ts(143,40): error TS2345: Argument of type 'string | undefined' is not assignable to parameter of type 'string'.
  Type 'undefined' is not assignable to type 'string'.
```

Here's my tsconfig:

```
{
  "compilerOptions": {
    "target": "ES2017",
    "module": "commonjs",
    "lib": [
      "es2017",
      "es2017.object"
    ],
    "sourceMap": true,
    "outDir": "./bin",
    "strict": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "noImplicitReturns": true,
    "noFallthroughCasesInSwitch": true,
    "moduleResolution": "node",
    "baseUrl": "./",
    "paths": {
      "*": ["./src/@types/*", "*"]
    },
    "pretty": true
  },
  "include": [
    "./src/**/*.*"
  ]
}
```

I'm not 100% sure what's causing it, because it doesn't actually happen when I build chrome-launcher locally, but I do know that there are 2 fixes:

1) The `canAccess` method already checks whether its argument is truthy, so we can update its signature to take a `string | null | undefined`.
2) The calls to the `RegExp` constructor can have a `!` appended to the environment variable to override the `strictNullChecks` tsconfig option for that expression.

--- a/.gitignore
+++ b/.gitignore
@@ -35,6 +35,7 @@ lighthouse-cli/*.js
 !chrome-launcher/compiled-check.js
 chrome-launcher/*.map
 chrome-launcher/*.js
+chrome-launcher/*.d.ts
 
 lighthouse-cli/commands/*.map
 lighthouse-cli/commands/*.js

--- a/chrome-launcher/.npmignore
+++ b/chrome-launcher/.npmignore
@@ -9,3 +9,9 @@ test/
 .eslintrc.js
 .travis.yml
 gulpfile.js
+
+# exclude source TypeScript files
+*.ts
+
+# allow TypeScript Declaration Files
+!*.d.ts

--- a/chrome-launcher/chrome-launcher.ts
+++ b/chrome-launcher/chrome-launcher.ts
@@ -7,12 +7,12 @@
 
 import * as childProcess from 'child_process';
 import * as fs from 'fs';
+import * as net from 'net';
+import * as rimraf from 'rimraf';
 import * as chromeFinder from './chrome-finder';
 import {getRandomPort} from './random-port';
 import {DEFAULT_FLAGS} from './flags';
 import {makeTmpDir, defaults, delay} from './utils';
-import * as net from 'net';
-const rimraf = require('rimraf');
 const log = require('lighthouse-logger');
 const spawn = childProcess.spawn;
 const execSync = childProcess.execSync;

--- a/chrome-launcher/package.json
+++ b/chrome-launcher/package.json
@@ -7,7 +7,7 @@
     "test": "mocha --require ts-node/register --reporter=dot test/**/*-test.ts --timeout=10000",
     "coverage": "nyc yarn test && nyc report --reporter=text-lcov > lcov.info",
     "test-formatting": "test/check-formatting.sh",
-    "format": "clang-format -i -style=file **/*.ts *.ts"
+    "format": "scripts/format.sh"
   },
   "devDependencies": {
     "@types/mocha": "^2.2.41",

--- a/chrome-launcher/package.json
+++ b/chrome-launcher/package.json
@@ -23,6 +23,7 @@
     "@types/core-js": "^0.9.41",
     "@types/mkdirp": "^0.3.29",
     "@types/node": "6.0.66",
+    "@types/rimraf": "^0.0.28",
     "lighthouse-logger": "^1.0.0",
     "mkdirp": "0.5.1",
     "rimraf": "^2.6.1"

--- a/chrome-launcher/scripts/format.sh
+++ b/chrome-launcher/scripts/format.sh
@@ -1,0 +1,5 @@
+#!/usr/bin/env bash
+
+FILES="`find . -type f \! -path '*node_modules*' \! -name '*.d.ts' -name '*.ts'`"
+
+./node_modules/.bin/clang-format -i -style=file $FILES

--- a/chrome-launcher/test/check-formatting.sh
+++ b/chrome-launcher/test/check-formatting.sh
@@ -12,5 +12,4 @@ check_formatting ()
   fi
 }
 
-check_formatting "*.ts"
-check_formatting "**/*.ts"
+check_formatting "`find . -type f \! -path '*node_modules*' \! -name '*.d.ts' -name '*.ts'`"

--- a/chrome-launcher/tsconfig.json
+++ b/chrome-launcher/tsconfig.json
@@ -2,6 +2,7 @@
     "compilerOptions": {
         "module": "commonjs",
         "target": "es2016",
+        "declaration": true,
         "noImplicitAny": true,
         "inlineSourceMap": true,
         "noEmitOnError": false,

